File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.py
def solution(x):
    sum=0
    str_x = str(x)
    for i in str_x:
        sum+=int(i)
    
    #이제 나누자
    if x %sum ==0:
        answer = True
    else:
        answer = False
    return answer
"""양의 정수 x가 하샤드 수라면 x의 자릿수 합으로 x가 나눠져야 합니다 
    그니까 일단 숫자가 들어와서 str로 변환을 해야 겠네
    그리고 for문을 통해 하나씩 빼온 다음에 이걸 다시 int로 만들어 하나씩 더해 
    그리고 다시 x를 나눠서 %== 0이면 true else false"""

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    var answer = true;
    let sum = String(x).split("").reduce((e,v)=> e+Number(v),0)
    answer = x%sum==0?true:false;
    return answer;
}
//하샤드 수 자릿수를 더한 값으로 나눠줘야 한다 그러면 일단 문자로 바꿔서 나눈 다음에 reduce넣기